### PR TITLE
Add a short summary of how shortcode OAuth works

### PR DESCRIPTION
**Pull request by EthanWaite**
Adds an explanation of how shortcode works to the OAuth reference page. Tried to keep this as short and concise as possible, referencing the REST API docs where possible for more info.

### CODE REVIEW

File: src/reference/oauth/index.pug [hunk 1]
```diff
@@ -117,6 +117,29 @@ block reference
...
+            #[a(href='https://beam.pro/go' target='_blank') beam.pro/go] and enter the #[code code].
+        li.
+            Your application polls
+            #[a(href='/rest.html#oauth_shortcode_check__handle__get' target='_blank') /oauth/shortcode/check/{handle}]
+            with the value of #[code handle] to check if the code has been used.
+        li.
+            If the user entered the code and accepted your application, you will receive a new
+            #[code code] which you will then pass to the
+            #[a(href='/rest.html#oauth_token_post' target='_blank') /oauth/token] endpoint through the
+            standard OAuth #[code authorization_code] process.
```
Comment (connor4312) [inline src/reference/oauth/index.pug]: Please don't manually wrap lines; put paragraphs on a single line and let editors do the wrapping.

File: src/reference/oauth/index.pug [hunk 2]
```diff
@@ -117,6 +117,29 @@ block reference
...
+        li.
+            Your application asks the user to go to
+            #[a(href='https://beam.pro/go' target='_blank') beam.pro/go] and enter the #[code code].
+        li.
+            Your application polls
+            #[a(href='/rest.html#oauth_shortcode_check__handle__get' target='_blank') /oauth/shortcode/check/{handle}]
+            with the value of #[code handle] to check if the code has been used.
+        li.
+            If the user entered the code and accepted your application, you will receive a new
+            #[code code] which you will then pass to the
```
Comment (connor4312) [inline src/reference/oauth/index.pug]: You don't get a new code, you get an OAuth Authorization Code. This is different than the six-digit code previously displayed to the user, we shouldn't conflate them.

File: src/reference/oauth/index.pug [hunk 3]
```diff
@@ -120,26 +120,16 @@ block reference
...
-            Your application polls
-            #[a(href='/rest.html#oauth_shortcode_check__handle__get' target='_blank') /oauth/shortcode/check/{handle}]
-            with the value of #[code handle] to check if the code has been used.
+            Your application polls #[a(href='/rest.html#oauth_shortcode_check__handle__get' target='_blank') /oauth/shortcode/check/{handle}] with the value of #[code handle] to check if the code has been used.
         li.
-            If the user entered the code and accepted your application, you will receive a new
-            #[code code] which you will then pass to the
-            #[a(href='/rest.html#oauth_token_post' target='_blank') /oauth/token] endpoint through the
-            standard OAuth #[code authorization_code] process.
+            If the user entered the code and accepted your application, you will receive an OAuth authorization code, #[code code], which you will then pass to the #[a(href='/rest.html#oauth_token_post' target='_blank') /oauth/token] endpoint through the standard #[code authorization_code] process.
```
Comment (connor4312) [inline src/reference/oauth/index.pug]: Last note: we can probably link the relevant section of the OAuth spec here

**Comment by EthanWaite** [comment]
Ooops, just noticed you asked for an example. What are you looking for there?

**Comment by ProbablePrime** [comment]
@EthanWaite A sample usually involves fully working through an item including examples of the responses retrieved.

@connor4312 It almost sounds like this would be better as a tutorial?

**Pull request merged**